### PR TITLE
[12.0][ADD] stock_location_template

### PR DESCRIPTION
**Pull request by AdriaGForgeFlow**
This PR introduces a new module stock_location_template in v12

## Description
This module introduces the concept of location template. In the template one
can define the number of cells (sub-location) that all locations with the same
template should have. It also allows to define a nomenclature that all locations
automatically generated from the template will follow.

## Usage
Go to Inventory > Configuration > Location Templates and Create your own
template.
Then go to Inventory > Configuration > Locations and select a specific template in
a location. New locations will be generated if the *Auto Generate Stock Locations*
option is selected in the template.

cc ~ @ForgeFlow @LoisRForgeFlow @JordiBForgeFlow

### CODE REVIEW

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Functional test LGTM

**Review by LoisRForgeFlow — COMMENTED** [review]
Looks great! Can you add a muted text in the form view of the location template explaining that you need to include a placeholder for the number in the name?

**Comment by AdriaGForgeFlow** [comment]
> Looks great! Can you add a muted text in the form view of the location template explaining that you need to include a placeholder for the number in the name?

Done! :+1:

**Comment by rousseldenis** [comment]
@AdriaGForgeFlow Do you plan to add tests to cover 100 % ?

File: stock_location_template/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2020 ForgeFlow S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Stock Location Template',
+    'summary': "Introduces the concept of location template",
+    'version': '12.0.1.0.0',
+    'license': 'AGPL-3',
+    'author': 'ForgeFlow S.L., Odoo Community Association (OCA)',
+    'website': 'https://forgeflow.com',
```
Comment (LoisRForgeFlow) [inline stock_location_template/__manifest__.py]: https://github.com/OCA/stock-logistics-warehouse/

File: stock_location_template/models/stock_location.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            if res.location_template_id.auto_generate_locations:
+                res._generate_sublocations()
+        return res
+
+    @api.multi
+    def write(self, vals):
+        for record in self:
+            location_template_id = vals.get("location_template_id", False)
+            if location_template_id and record.location_template_id:
+                    record._check_can_change_template()
```
Comment (LoisRForgeFlow) [inline stock_location_template/models/stock_location.py]: you can remove an identation here right?

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM, very small comments

**Comment by LoisRForgeFlow** [comment]
@rousseldenis test coverage increased!

/ocabot merge nobump

**Comment by rousseldenis** [comment]
@LoisRForgeFlow In the last changes of ocabot merge, the option is mandatory. So, we could be tempted to use nobump for migrations. Don't use it as it won't generate wheels for pypi. Use patch instead. Even if it lead to 12.0.1.0.1 version number.

**Comment by LoisRForgeFlow** [comment]
@rousseldenis  even if is an ADD? how is it that wheels are not generated for new modules?

**Comment by rousseldenis** [comment]
That's the same. The idea was to require options. See https://github.com/OCA/oca-github-bot/issues/90

The nobump is used when you make fast merges and then a release one for example.

In most cases, use patch|minor|major

**Comment by LoisRForgeFlow** [comment]
Ok, quite ugly to be forced to start in 1.0.1 but anyway not a big deal. I stopped merge and restart now:

/ocabot merge patch

**Comment by LoisRForgeFlow** [comment]
Seems it got stuck:

/ocabot merge patch

**Pull request merged**